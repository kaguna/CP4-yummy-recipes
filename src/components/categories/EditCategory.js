/**
 * This is a modal component to edit a category name and the procedure.
 */
import React, { Component } from 'react';
import axiosInstance from '../../common/AxiosInstance';

class EditCategory extends Component {
  constructor(props) {
    super(props);
    this.state = { errorMessage: '', category_name: '' };
  }
    inputHandler = (event) => {
      event.preventDefault();
      const { name, value } = event.target;
      this.setState({ [name]: value });
    };

    editCategoryHandler = (event) => {
      event.preventDefault();
      const { category_name } = this.state;
      axiosInstance.put(`/category/${this.props.category.id}`, { category_name })
        .then((response) => {
          this.props.parent.setState({ successMessage: response.data.message, errorMessage: '' });
          this.props.categoryAfterEdit();
        })
        .catch((error) => {
          if (error.response) {
            this.setState({ errorMessage: error.response.data.message });
          } else if (error.request) {
            this.setState({ errorMessage: 'Cant connect to the server.Please check your connection and try again.' });
          }
        });
    };
    render() {
      const { category_id } = this.props;
      const { category_name } = this.props.category;
      const { errorMessage } = this.state;
      return (

        <div
          className="modal fade"
          id={category_id}
          tabIndex="-1"
          role="dialog"
          aria-labelledby="exampleModalCenterTitle"
          aria-hidden="true"
        >
          <div className="modal-dialog" role="document">
            <div className="modal-content">
              <div className="modal-header">
                <h5 className="modal-title" id="exampleModalLongTitle">Edit category ({category_name})</h5>
              </div>
              <div className="modal-body">
                {errorMessage ?
                  <div className="alert alert-danger">{errorMessage}</div> : ''}

                <div className="form-group">
                  <label className="control-label col-sm-3">New Category Name:</label>
                  <div className="col-sm-8">
                    <input
                      type="text"
                      name="category_name"
                      className="form-control"
                      placeholder="Enter New Category name"
                      defaultValue={category_name}
                      onChange={this.inputHandler}
                    />
                  </div>
                </div>
              </div>
              <div className="modal-footer">
                <button className="btn btn-danger" data-dismiss="modal">Close</button>
                <button className="btn btn-primary" data-dismiss="modal" onClick={this.editCategoryHandler}>
                  <i className="glyphicon glyphicon-edit" /> Edit Category
                </button>
              </div>

            </div>
          </div>
        </div>
      );
    }
}
export default EditCategory;
