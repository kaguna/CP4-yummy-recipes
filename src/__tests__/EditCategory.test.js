// This file tests category editing with the properties.
import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import EditCategory from '../components/categories/EditCategory';

describe('Login component', () => {
  const props = {
    category: { category_name: 'muthere' },
  };
  const wrapper = shallow(<EditCategory {...props} />);

  it('renders properly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
