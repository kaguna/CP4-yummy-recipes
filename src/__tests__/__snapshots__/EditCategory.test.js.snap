// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders properly 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="exampleModalCenterTitle"
  className="modal fade"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
          id="exampleModalLongTitle"
        >
          Edit category (
          muthere
          )
        </h5>
      </div>
      <div
        className="modal-body"
      >
        <div
          className="form-group"
        >
          <label
            className="control-label col-sm-3"
          >
            New Category Name:
          </label>
          <div
            className="col-sm-8"
          >
            <input
              className="form-control"
              defaultValue="muthere"
              name="category_name"
              onChange={[Function]}
              placeholder="Enter New Category name"
              type="text"
            />
          </div>
        </div>
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-danger"
          data-dismiss="modal"
        >
          Close
        </button>
        <button
          className="btn btn-primary"
          onClick={[Function]}
        >
          <i
            className="glyphicon glyphicon-edit"
          />
           Edit Category
        </button>
      </div>
    </div>
  </div>
</div>
`;
